server/util/capabilities/capabilities.go

```diff
@@ -57,3 +57,19 @@ func IsGranted(ctx context.Context, env environment.Env, cap akpb.ApiKey_Capabil
 	}
 	return user.HasCapability(cap), nil
 }
+
+func GetFromContext(ctx context.Context, env environment.Env) ([]akpb.ApiKey_Capability, error) {
+	var caps []akpb.ApiKey_Capability
+	if auth := env.GetAuthenticator(); auth != nil {
+		u, err := auth.AuthenticatedUser(ctx)
+		if err == nil {
+			caps = u.GetCapabilities()
+		} else {
+			if perms.IsAnonymousUserError(err) && auth.AnonymousUsageEnabled() {
+				caps = DefaultAuthenticatedUserCapabilities
+			}
+		}
+	}
```

Comment:
nit: can use early returns here to reduce nesting, and also return an error in the case where there is a legitimate auth error. (In the invocationdb callsite, you can ignore the returned error, since that's what the previous code was doing)

```go
auth := env.GetAuthenticator()
if auth == nil {
  return nil, status.UnimplementedError("auth is not configured")
}
u, err := auth.AuthenticatedUser(ctx)
if err != nil {
  if perms.IsAnonymousUserError(err) && auth.AnonymousUsageEnabled() {
    return DefaultAuthenticatedUserCapabilities, nil
  }
  return nil, err
}
return u.GetCapabilities(), nil
```

Reply:
Ack - done